### PR TITLE
Homework12: Alexey Kozhan

### PR DESCRIPTION
**Pull request by AlexKozhan**
Homework12 is done in 3 files

### CODE REVIEW

File: cards.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        self.cards.append(Card('Black', 'Joker'))
+        self.cards.append(Card('Red', 'Joker'))
+
+    def shuffle(self):
+        """Shuffles the deck of cards."""
+        random.shuffle(self.cards)
+
+    def get(self, index):
+        """Returns the card at the specified index."""
+        if 0 <= index < len(self.cards):
```
Comment (Stml89) [inline cards.py:52]: Better to move this check `0 < X < len()` to where you ask user for a number

File: converter.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+assert bank.exchange_currency(petya.currency,
+ petya.amount, 'USD') == (10.76, "USD"), <error message>"""
+
+
+class Bank:
+    """
+    Represents a bank that performs currency exchange.
+    """
+
+    # pylint: disable=too-few-public-methods
```
Comment (Stml89) [inline converter.py:31]: Please, use `.pylintrc` file

File: himiya.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+from typing import Dict
+
+
+class Himiya:
+    """
+    A class to parse a chemical formula
+    and count the number of each type of atom.
+    """
+
+    def __init__(self):
```
Comment (Stml89) [inline himiya.py:31]: do you really need this method?

File: himiya.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+                    num = 1 if j == k else int(formula[j:k])
+                    current[atom] += num
+                    i = k
+
+        self.add_remaining_to_stack(stack, current)  # Add
+        # remaining counts to the top of stack
+
+        return dict(sorted(current.items()))
+
+    def multiply_current(self, current: Dict[str, int], num: int):
```
Comment (Stml89) [inline himiya.py:80]: Use `@staticmethod`

File: himiya.py [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+повторяется каждый атом внутри скобок.
+Индекс после скобки не обязателен. Если его нет, значит
+содержимое скобок повторяется 1 раз."""
+
+from collections import defaultdict
+import copy
+from typing import Dict
+
+
+class Himiya:
```
Comment (Stml89) [inline himiya.py:25]: Chemistry

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: cards.py (hunk 1 above)
Comment (AlexKozhan) [reply cards.py]: fixed

File: converter.py (hunk 2 above)
Comment (AlexKozhan) [reply converter.py:31]: fixed

File: himiya.py (hunk 3 above)
Comment (AlexKozhan) [reply himiya.py:31]: there was pylint too-few public methods error again as in homework11. Created .pylintrc file - next time i will not create such methods. so - fixed.

File: himiya.py (hunk 4 above)
Comment (AlexKozhan) [reply himiya.py]: fixed

File: himiya.py (hunk 5 above)
Comment (AlexKozhan) [reply himiya.py]: fixed

**Comment by Stml89** [comment]
Please, merge ALL previous PRs

**Review by Stml89 — APPROVED** [review]

**Comment by AlexKozhan** [comment]
> Please, merge ALL previous PRs
i will

**Pull request merged**